presto-native-execution/presto_cpp/main/http/tests/HttpTest.cpp

```diff
@@ -133,7 +133,7 @@ class HttpClientFactory {
       const folly::SocketAddress& address,
       const std::chrono::milliseconds& timeout) {
     return std::make_unique<http::HttpClient>(
-        eventBase_.get(), address, timeout);
+        eventBase_.get(), address, nullptr, timeout);
```

Comment:
Can you add a test that record something in statsReportFunc and check it has been recorded in test?
```
std::atomic<int> reportedCount = 0;
[&]() { ++reportedCount; } as report stats function

At the end of test
ASSERT_EQ(reportedCount, numHttpChains);
```